cpp/test/Slice/errorDetection/TypeAlias.ice

```diff
@@ -0,0 +1,80 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+module Test
+{
+    sequence<float> floatSeq;
+
+    struct Custom
+    {
+        int i;
+    }
+
+    class C0 {}
+    interface I0 {}
+    exception E0 {}
+
+    module Nested1
+    {
+        class C1;
+    }
+
+    module Nested2
+    {
+        class C2;
+    }
+
+    // Basic type aliases
+    using intAlias = int;               //ok
+    using stringAlias = string;         //ok
+    using floatSeqAlias = floatSeq;     //ok
+    using customAlias = Custom;         //ok
+    using unknownAlias = foo;           //can't resolve type
+    using optintAlias = int?;           //can't typealias optional types
+    using missingType;                  //missing underlying type
+    using aliasName = int               //missing semicolon
+
+    using C0Alias = C0;                      //ok
+    using C1Alias = C1;                      //can't resolve type
+    using scopedC0Alias = Test::C0;          //ok
+    using scopedC1Alias = Test::Nested1::C1; //ok
+
+    // Type resolution of type-aliases
+    module Nested2
+    {
+        using C0Alias = C0;                      //ok
+        using C1Alias = C1;                      //can't resolve type
+        using C2Alias = C2;                      //ok
+        using scopedC0Alias = Test::C0;          //ok
+        using scopedC1Alias = Test::Nested1::C1; //ok
+        using scopedC2Alias = Test::Nested2::C2; //ok
+    }
+
+    // Inheritance from type-aliases
+    // Note, all metadata is allowed, but non-type metadata will error out when the alias is used.
+    using C0Alias = C0;                            //redefinition error
+    using C0AliasMetadata = [protected] C0;        //ok
```

Comment:
I don't see how this will work, the only generated class is C0, what is the purpose of applying metadata to the alias?

Reply:
From a Slice perspective, you're right, these aliases are useless.
From a test perspective though, these are here to make sure that aliases with metadata are parsed correctly and don't raise any unexpected errors. It's just for testing purposes.

Reply:
I agree that type-like metadata should be an error, but right now we don't have the ability to distinguish between different types of metadata. I'll tighten these checks once I've had a chance to add some structure to how we handle metadata.